### PR TITLE
feat: initial implementation for the operations code generation

### PR DESCRIPTION
**Pull request by jensneuse**
## Motivation and Context

Users want to be able to quickly turn a data source into API Endpoints.
Using auto-generated operations, they can generate (customizable) Operations with as little as 2 lines of code. (see docs)

## Marketing Changelog

We've got feedback from our user base that it can be tedious to import a large amount of OpenAPI Endpoints and then have to write out every Operation, although 90% of all Operations can be exposed as is.

With auto-generated Operations, you can filter API Namespaces and root fields in 2 lines of code, and WunderGraph will generate Operations for all selected fields automatically. And even if the generated Operation is not exactly what you needed, there's an easy way to customize it.

### CODE REVIEW

File: docs-website/src/pages/docs/auto-generated-operations-reference/index.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+## Example configuration
+
+```typescript
+// .wundergraph/wundergraph.generate.ts
+import { configureWunderGraphGeneration } from '@wundergraph/sdk';
+
+export default configureWunderGraphGeneration({
+  codeGenerators: [], // abbreviated for brevity
+  operationsGeneration: (config) => {
```
Comment (Pagebakers) [inline docs-website/src/pages/docs/auto-generated-operations-reference/index.md:35]: `operationsGenerator` would be more consistent with `codeGenerators`

File: docs-website/src/pages/docs/auto-generated-operations-reference/index.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  apis: [weather, federated],
+  generate,
+});
+```
+
+## Customization of auto-generated Operations
+
+Here's an example of how a generated Operation looks like:
+
+```graphql
```
Comment (Pagebakers) [inline docs-website/src/pages/docs/auto-generated-operations-reference/index.md:87]: docs don't mention where the operation is generated, in the root of operations/ or in a sub directory eg operations/namespace?

**Review by Pagebakers — CHANGES_REQUESTED** [review]
Massive productivity gain, a few comments on the docs.

Also should mention this in the wundergraph.config.ts reference.

File: docs-website/src/pages/docs/auto-generated-operations-reference/index.md (hunk 1 above)
Comment (fiam) [reply docs-website/src/pages/docs/auto-generated-operations-reference/index.md:35]: I'd say these are different things, I think there don't need to be named in the same way.

File: docs-website/src/pages/docs/auto-generated-operations-reference/index.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+export default configureWunderGraphGeneration({
+  codeGenerators: [], // abbreviated for brevity
+  operationsGeneration: (config) => {
+    // include all root fields from the "weather" & "federated" API Namespace
+    config.includeNamespaces('weather', 'federated');
+    // we could also exclude the root field "me" from the "federated" API Namespace
+    config.filterRootFields(
+      (field) => field.operationType === 'query' && field.apiNamespace === 'federated' && field.name === 'me'
+    );
```
[the remark this answers is not in the document]
Comment (jensneuse) [reply docs-website/src/pages/docs/auto-generated-operations-reference/index.md:41]: yes

File: docs-website/src/pages/docs/auto-generated-operations-reference/index.md (hunk 1 above)
Comment (jensneuse) [reply docs-website/src/pages/docs/auto-generated-operations-reference/index.md:35]: changed

**Comment by jensneuse** [comment]
> Massive productivity gain, a few comments on the docs.
> 
> Also should mention this in the wundergraph.config.ts reference.

Once released, I'll write a guide and we can also update some examples.

**Review by Pagebakers — APPROVED** [review]

**Pull request merged**